### PR TITLE
feat: support schema registry in risedev

### PR DESCRIPTION
**Pull request by xxchan**
Signed-off-by: xxchan <xxchan22f@gmail.com>I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
related: #16903, prepare for #16931

Besides, don't use redpanda schema registry in ci to reduce confusion.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: e2e_test/source/basic/nosim_kafka.slt [hunk 1]
```diff
@@ -1,3 +1,6 @@
+control substitution on
```
Comment (xxchan) [inline e2e_test/source/basic/nosim_kafka.slt:1]: Only try `RISEDEV_SCHEMA_REGISTRY_URL` in this test. Not bother to modify others.

File: src/risedevtool/src/bin/risedev-dev.rs [hunk 2]
```diff
@@ -279,6 +279,18 @@ fn task_main(
                 ctx.pb
                     .set_message(format!("kafka {}:{}", c.address, c.port));
             }
+            ServiceConfig::SchemaRegistry(c) => {
+                let mut ctx =
+                    ExecuteContext::new(&mut logger, manager.new_progress(), status_dir.clone());
+                let mut service = SchemaRegistryService::new(c.clone());
+                service.execute(&mut ctx)?;
+                let mut task =
+                    risedev::TcpReadyCheckTask::new(c.address.clone(), c.port, c.user_managed)?;
```
Comment (BugenZhao) [inline src/risedevtool/src/bin/risedev-dev.rs:288]: Have you checked whether this is accurate?

File: e2e_test/source/basic/nosim_kafka.slt [hunk 3]
```diff
@@ -1,3 +1,6 @@
...
+
+# FIXME: does this really work??
```
Comment (BugenZhao) [inline e2e_test/source/basic/nosim_kafka.slt:3]: 😄

File: src/risedevtool/src/task/schema_registry_service.rs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            ),
+        ]
+    }
+
+    fn ports(&self) -> Vec<(String, String)> {
+        vec![(self.port.to_string(), "8081".to_owned())]
+    }
+
+    fn data_path(&self) -> Option<String> {
+        None
```
Comment (BugenZhao) [inline src/risedevtool/src/task/schema_registry_service.rs:60]: Shall we support persisting data?

**Review by BugenZhao — APPROVED** [review]
Clever

File: src/risedevtool/src/task/schema_registry_service.rs (hunk 4 above)
Comment (xxchan) [reply src/risedevtool/src/task/schema_registry_service.rs:60]: Data is stored in kafka

File: src/risedevtool/src/bin/risedev-dev.rs (hunk 2 above)
Comment (xxchan) [reply src/risedevtool/src/bin/risedev-dev.rs:288]: How can I check?

Comment (tabVersion) [reply src/risedevtool/src/bin/risedev-dev.rs:288]: maybe perform a GET request and check the code equals to 200. Here's an example

```
curl -X GET http://localhost:8081/subjects
```

**Pull request merged**